http/transport.go

```diff
@@ -668,6 +671,64 @@ func invokeGenerateKeys(client *http.Client, t flux.Token, router *mux.Router, e
 	return nil
 }
 
+func handlePostIntegrationsGithub(s api.FluxService) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var (
+			inst  = getInstanceID(r)
+			vars  = mux.Vars(r)
+			owner = vars["owner"]
+			repo  = vars["repository"]
+			tok   = r.Header.Get("GithubToken")
+		)
+
+		if repo == "" || owner == "" || tok == "" {
+			w.WriteHeader(http.StatusUnprocessableEntity)
+			fmt.Fprintf(w, "repo, owner or token is empty")
+			return
+		}
+
+		// Generate deploy key
+		err := s.GenerateDeployKey(inst)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, err.Error())
+			return
+		}
+
+		// Obtain the generated key
+		cfg, err := s.GetConfig(inst)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, err.Error())
+			return
+		}
+		publicKey := cfg.Git.HideKey().Key
+
+		// Use the Github API to insert the key
+		// Have to create a new instance here because there is no
+		// clean way of injecting without significantly altering
+		// the initialisation (at the top)
+		gh := git.Github{
+			Client: http.Client{
+				Timeout: 10 * time.Second,
+			},
+		}
+		err = gh.InsertDeployKey(inst, owner, repo, tok, publicKey)
+		if err != nil {
+			httpErr, isHttpErr := err.(*httperror.APIError)
+			if isHttpErr {
+				w.WriteHeader(httpErr.StatusCode)
```

Comment:
Do we definitely want to pass on exactly the code GitHub returned? (Does this _happen_ to be correct, or is it a shortcut ..)